### ISSUE TITLE
Split MMSpec and MLSpec from QCSpec

### ISSUE THREAD

**Issue by jthorton** [openforcefield/openff-qcsubmit#152]
To allow for more specific validation of calculation keywords and settings it would be good to split the current QCSpec class into 3 separate objects with their own validation. `QCSpec` for QM calculations `MMSpec` for MM calculations and `MLSpec` for ML calculations.

**Comment by mattwthompson** [comment]
How many programs would this be? (Not shooting the idea down, just need to be reminded what "program" means in this context)

**Comment by j-wags** [comment]
Without knowing much about the specifics of this ask, when I was refactoring qcsubmit a lot of the pain was caused by the fact that many QCS objects were just wrappers around QCF/QCEl objects. This meant that, when the QCF/QCEl objects changed, the QCS wrappers needed to change in lockstep.

So I'm nervous about this potentially putting more layers of complexity/behavior between the QCSubmit API and the underlying QCEl interface. Like, if we helpfully validated user input and raised a validation error if they tried to ask ORCA for MBIS charge, and then one day ORCA added support for MBIS charges, we'd need to hustle to make a release to get the validation up to date. It'd be this way for every feature that we wrap/validate. 

Is there a clear sweet spot for user experience versus maintenance cost for this sort of thing?

**Comment by jthorton** [comment]
>  Like, if we helpfully validated user input and raised a validation error if they tried to ask ORCA for MBIS charge, and then one day ORCA added support for MBIS charges, we'd need to hustle to make a release to get the validation up to date. 

Yeah that was always the plan with qcsubmit to have strict validation on the programs we are interested in and then everything else we just let through see [here](https://github.com/openforcefield/openff-qcsubmit/blob/23fde57cf6a8e00a6a7c4ac3ca4d5be7947fd6c9/openff/qcsubmit/common_structures.py#L457) and even then it's only for a subset of features which we often use. Ideally each program would tell use what features are available before hand but outside of that I am not sure I see many other options but open to ideas as I would like to get this or someting similar working for bespokefit.
